src/promptflow/promptflow/_sdk/_telemetry/logging_handler.py

```diff
@@ -73,71 +52,104 @@ def get_scrubbed_cloud_role():
+            _APPLICATION_INSIGHTS_EVENT_MARKER_ATTRIBUTE: True,
+            **custom_dimensions,
+        }
+        envelope = super()._log_to_envelope(log_data=log_data)
         # scrub data before sending to appinsights
         role = get_scrubbed_cloud_role()
         envelope.tags["ai.cloud.role"] = role
         envelope.tags.pop("ai.cloud.roleInstance", None)
         envelope.tags.pop("ai.device.id", None)
         return envelope
 
+    def _should_collect_stats(self):
+        return False
+
+
+# cspell:ignore AzureMLSDKLogHandler
+class PromptFlowSDKLogHandler(LoggingHandler):
+    """Customized AzureLogHandler for PromptFlow SDK"""
+
+    def __init__(self, custom_dimensions, enable_telemetry):
+        self._is_telemetry_enabled = enable_telemetry
+        self.disable_telemetry_logger()
+        logger_provider = LoggerProvider()
+        exporter = PromptFlowSDKExporter(
+            connection_string=f"InstrumentationKey={INSTRUMENTATION_KEY}",
+            custom_dimensions=custom_dimensions,
+        )
+        logger_provider.add_log_record_processor(BatchLogRecordProcessor(exporter))
+        super().__init__(logger_provider=logger_provider)
+
+    def emit(self, record: logging.LogRecord):
+        # skip logging if telemetry is disabled
+        if not self._is_telemetry_enabled:
+            return
+
+        try:
+            super().emit(record)
+            # log the record immediately if it is an error
+            if record.exc_info and not all(item is None for item in record.exc_info):
+                self.flush()
+        except Exception:  # pylint: disable=broad-except
+            # ignore any exceptions, telemetry collection errors shouldn't block an operation
+            return
+
     @classmethod
     def disable_telemetry_logger(cls):
-        """Disable AzureEventHandler's logging to avoid warning affect user experience"""
-        from opencensus.ext.azure.common.processor import logger as processor_logger
-        from opencensus.ext.azure.common.storage import logger as storage_logger
-        from opencensus.ext.azure.common.transport import logger as transport_logger
+        """Disable opentelemetry's logging to avoid warning affect user experience"""
+
+        from azure.monitor.opentelemetry.exporter.export.logs._exporter import _logger as exporter_logger
+        from opentelemetry._logs._internal import _logger as provider_logger
+        from opentelemetry.sdk._logs._internal import _logger as handler_logger
+        from opentelemetry.sdk._logs._internal.export import _logger as processor_logger
 
+        handler_logger.setLevel(logging.CRITICAL)
         processor_logger.setLevel(logging.CRITICAL)
-        transport_logger.setLevel(logging.CRITICAL)
-        storage_logger.setLevel(logging.CRITICAL)
+        provider_logger.setLevel(logging.CRITICAL)
+        exporter_logger.setLevel(logging.CRITICAL)
+
+
+@cache_result_with_expire(result_type=PromptFlowSDKLogHandler)
```

Comment:
why do we need the expire thing here?

Reply:
I want to use this function as a universal function, so in some scenarios, the cached function results may have an expiration date.

Reply:
why not using python builtin cache macanism like functools.lru_cache?

Reply:
Just to customize some features, such as specifying the type of return, I only cache results of type **PromptFlowSDKLogHandler**, and I don't need to cache results of type **logging.NullHandler**. 
I hope to try again to get the correct result when calling this function the next time.

![image](https://github.com/microsoft/promptflow/assets/23182548/a1aa60a7-4b14-40fc-b227-68af2287fe35)

Reply:
I deleted the expiration time, which doesn't seem to be very useful.